app/models/piece.rb

```diff
@@ -1,2 +1,45 @@
 class Piece < ApplicationRecord
+	
+	Pieces.each do |piece|
+		if piece.y_position == start_y && piece.x_position.between?(start_x, dest_x)
+				return false
+			end
+		end
+		return true
+	end
+
+	def long_check?(start_x,start_y,dest_x,dest_y)
+		Pieces.each do |piece|
+			if piece.x_position == start_x && piece.y_position.between?(start_y, dest_y)
+				return false
+			end
+		end
+		return true
+	end
```

Comment:
@Kevlau87 what is long_check? Suggest to add some comments to brief describe the function.

Reply:
outdated commit